### PR TITLE
Consumable: Support GDPR and US Privacy consent

### PR DESCRIPTION
**Pull request by djcsdy**
This change passes GDPR and US Privacy consent information through to Consumable.

I am proposing this change on behalf of Consumable, cc @naffis.

### CODE REVIEW

File: adapters/consumable/consumable.go [hunk 1]
```diff
@@ -18,6 +18,15 @@ type ConsumableAdapter struct {
 	endpoint string
 }
 
+type regsExt struct {
```
Comment (SyntaxNode) [inline adapters/consumable/consumable.go:21]: These structures are already defined as `openrtb_ext.ExtRegs` and `openrtb_ext.ExtUser`. Please see my other comment on helper code you can use or consider using the shared structures.

File: adapters/consumable/consumable.go [hunk 2]
```diff
@@ -124,6 +140,29 @@ func (a *ConsumableAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
 		body.Url = request.Site.Page     // where the impression will be made
 	}
 
+	gdpr := bidGdpr{}
+
+	if request.Regs != nil && request.Regs.Ext != nil {
```
Comment (SyntaxNode) [inline adapters/consumable/consumable.go:145]: You may want to consider using `ccpa.ReadPolicy()` in the privacy package.

File: adapters/consumable/consumable.go [hunk 3]
```diff
@@ -124,6 +140,29 @@ func (a *ConsumableAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
...
+
+			if regsExt.Gdpr != nil {
+				applies := *regsExt.Gdpr != 0
+				gdpr.Applies = &applies
+				body.Gdpr = &gdpr
+			}
+		}
+	}
+
+	if request.User != nil && request.User.Ext != nil {
```
Comment (SyntaxNode) [inline adapters/consumable/consumable.go:158]: I'll eventually have a similar `gdpr.ReadPolicy()` at some point, but that's still a few months away. I'll ping you (maybe via a proposed PR?) to streamline this code when that's done. For now, please consider using the shared structs.

File: adapters/consumable/consumable.go [hunk 4]
```diff
@@ -32,6 +41,8 @@ type bidRequest struct {
 	Url                string      `json:"url,omitempty"`
 	EnableBotFiltering bool        `json:"enableBotFiltering,omitempty"`
 	Parallel           bool        `json:"parallel"`
+	Ccpa               string      `json:"ccpa,omitempty"`
```
Comment (SyntaxNode) [inline adapters/consumable/consumable.go:44]: Nitpick: The standard naming in Go is to use uppercase letters for acronyms. Please consider `CCPA` and `GDPR`.

File: adapters/consumable/consumable.go (hunk 1 above)
Comment (djcsdy) [reply adapters/consumable/consumable.go:21]: Thanks for the heads up. I did search for existing definitions but I couldn’t find them. Not sure how I missed them, too many occurrences of `us_privacy` and `gdpr` to dig through perhaps :-).

File: adapters/consumable/consumable.go (hunk 3 above)
Comment (djcsdy) [reply adapters/consumable/consumable.go:158]: Will do, please do ping me when `gdpr.ReadPolicy()` is available.

File: adapters/consumable/consumable.go (hunk 1 above)
Comment (SyntaxNode) [reply adapters/consumable/consumable.go:21]: Perhaps. :)

File: adapters/consumable/consumable.go (hunk 3 above)
Comment (SyntaxNode) [reply adapters/consumable/consumable.go:158]: Aye.

**Comment by djcsdy** [comment]
@SyntaxNode Thanks for your suggestions. I have pushed changes to address your comments.

**Pull request merged**